src/main/kotlin/uk/gov/justice/digital/hmpps/deliusapi/service/throughcare/RecallService.kt

```diff
@@ -177,12 +179,67 @@ class RecallService(
 
     val prisonOffenderManager = PrisonOffenderManager(
       offenderId = offenderId,
-      providerId = provider.id,
-      teamId = team.id,
-      staffId = staff.id,
+      provider = provider,
+      team = team,
+      staff = staff,
       allocationDate = LocalDateTime.now(),
-      allocationReasonId = allocationReason.id
+      allocationReason = allocationReason
     )
-    prisonOffenderManagerRepository.save(prisonOffenderManager)
+    return prisonOffenderManagerRepository.save(prisonOffenderManager)
+  }
+
+  private fun updateResponsibleOfficer(offender: Offender, newPrisonOffenderManager: PrisonOffenderManager) {
+    val currentRO = responsibleOfficerRepository.findByOffenderId(offender.id)
+
+    // End date previous responsible officer
+    if (currentRO != null) {
+      currentRO.endDate = LocalDateTime.now()
+      responsibleOfficerRepository.save(currentRO)
+    }
+
+    // Create new responsible officer linked to new POM
+    val newResponsibleOfficer = ResponsibleOfficer(
+      offender = offender,
+      prisonOffenderManager = newPrisonOffenderManager,
+      startDate = LocalDateTime.now()
+    )
+    responsibleOfficerRepository.save(newResponsibleOfficer)
+
+    var notes = listOfNotNull(
+      "New Details:",
+      "Responsible Officer Type: Prison Offender Manager",
+      "Responsible Officer: ${newPrisonOffenderManager.staff.getDisplayName()} " +
+        "(${newPrisonOffenderManager.team.description}, " +
+        "${newPrisonOffenderManager.provider.description})",
+      "Start Date: ${newResponsibleOfficer.startDate.format(DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm:ss"))}",
+      "Allocation Reason: ${newPrisonOffenderManager.allocationReason.description}"
+    ).joinToString("\n")
```

Comment:
Minor thing, but a multi-line string might be cleaner here e.g.
```suggestion
    var notes = """
      New Details:
      Responsible Officer Type: Prison Offender Manager
      Responsible Officer: ${newPrisonOffenderManager.staff.getDisplayName()} (${newPrisonOffenderManager.team.description}, ${newPrisonOffenderManager.provider.description})
      Start Date: ${newResponsibleOfficer.startDate.format(DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm:ss"))}
      Allocation Reason: ${newPrisonOffenderManager.allocationReason.description}
    """.trimIndent()
```